src/App.js

```diff
@@ -0,0 +1,316 @@
+          ? {
+              ...list,
+              listTasks: list.listTasks.map((task) =>
+                task.id === taskId
+                  ? {
+                      ...task,
+                      taskName: taskName,
+                      taskDate: taskDate,
+                      taskTime: taskTime,
+                      taskNotes: taskNotes,
+                      isTaskCompleted: taskStatus,
+                    }
+                  : task
+              ),
+            }
+          : list
+      )
+    );
+  }
+
+  function handleDeleteTask(listId, taskId) {
+    setData(
+      data.map((list) =>
+        list.id === listId
+          ? {
+              ...list,
+              listTasks: list.listTasks.filter((task) => task.id !== taskId),
+            }
+          : list
+      )
+    );
+    setCurrentPage(
+      "SingleListPage"
+    ); /* after deleting task, redirect to Single List Page */
+  }
+
+  function handleEditList(listId, listField, newValue) {
+    setData(
+      data.map((list) =>
+        list.id === listId ? { ...list, [listField]: newValue } : list
+      )
+    );
+  }
+
+  function handleEditListAppearance(listId, newName, newIcon) {
+    setData(
+      data.map((list) =>
+        list.id === listId
+          ? { ...list, listName: newName, listIcon: newIcon }
+          : list
+      )
+    );
+  }
+
+  function handleDeleteCompletedTasks(listId) {
+    setData(
+      data.map((list) =>
+        list.id === listId
+          ? {
+              ...list,
+              listTasks: list.listTasks.filter((task) => !task.isTaskCompleted),
+            }
+          : list
+      )
+    );
+  }
+
+  function handleDeleteAllTasks(listId) {
+    setData(
+      data.map((list) =>
+        list.id === listId ? { ...list, listTasks: [] } : list
+      )
+    );
+  }
+
+  function handleDeleteList(listId) {
+    setData(data.filter((list) => list.id !== listId));
+    setCurrentPage("Home"); /* after deleting list, redirect to Home Page */
+  }
+
+  // Code below changes current/previous page, list, and task
+  const [currentPage, setCurrentPage] = useState("Home");
+  const [prevPage, setPrevPage] = useState("Home");
+  const [currentListId, setCurrentListId] = useState();
+  const [currentTaskId, setCurrentTaskId] = useState();
+
+  function handleChangePage(newPage) {
+    setPrevPage(currentPage);
+    setCurrentPage(newPage);
+    if (newPage === "Home") {
+      handleChangeList(null);
+    } else if (newPage === "SingleListPage") {
+      handleChangeTask(null);
+    }
+  }
+
+  function handleChangeList(newListId) {
+    setCurrentListId(newListId);
+  }
+
+  function handleChangeTask(newTaskId) {
+    setCurrentTaskId(newTaskId);
+  }
+
+  // Functions below handle list and task creation
+  function handleCreateList(listName, listIcon) {
+    const newList = {
+      id: generateUniqueID(),
+      listName: listName,
+      listIcon: listIcon,
+      areCompletedTasksHidden: false,
```

Comment:
Interesting that this is permanent state (in the list) rather than transitory. Do you expect you'll save this in Firestore?